### PR TITLE
feat: Change middleware to also handle AuthV2

### PR DESCRIPTION
**Pull request by chanioxaris**
## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🌟 New feature (non-breaking change which adds functionality)
- [ ] 🔨 Breaking change (fix or feature that would cause existing functionality)
- [ ] 📖 Docs change / refactoring / dependency upgrade to change)

## Description

This PR changes the current middleware `WithSession` to also handle session tokens that will be used in the upcoming networkless authentication. If successful adds the JWT claims in the request context so you can easily access them inside a handler. This change should not affect or change the middleware behavior for the current use cases

### CODE REVIEW

File: clerk/middleware.go [hunk 1]
```diff
@@ -3,30 +3,63 @@ package clerk
...
+					return
+				}
+
+				ctx := context.WithValue(r.Context(), ActiveSessionClaims, claims)
+				next.ServeHTTP(w, r.WithContext(ctx))
+			} else {
+				// Validate using session verify request
+				session, err := client.Verification().Verify(r)
+				if err != nil {
+					w.WriteHeader(http.StatusUnauthorized)
```
Comment (chanioxaris) [inline clerk/middleware.go:32]: Before we returned a 400 Bad Request if the verification failed. Do you think we should change it to a 401 Unauthorized as more appropriate status code?

File: clerk/clerk.go [hunk 2]
```diff
@@ -26,8 +26,8 @@ type Client interface {
 	NewRequest(method string, url string, body ...interface{}) (*http.Request, error)
 	Do(req *http.Request, v interface{}) (*http.Response, error)
 
-	DecodeToken(token string) (*Claims, error)
-	VerifyToken(token string) (*Claims, error)
+	DecodeToken(token string) (*TokenClaims, error)
+	VerifyToken(token string) (*SessionClaims, error)
```
Comment (agis) [inline clerk/clerk.go:30]: ❓ Isn't this a breaking change? Or do these new types have the same API as the previous one? (Even so, it's still breaking for users that checked the actual type).

Comment (chanioxaris) [reply clerk/clerk.go:30]: After we add these function utilities, we haven't released a new sdk version so i doubt anyone are using them at all

Comment (agis) [reply clerk/clerk.go:30]: Yeah forgot that 😊

File: clerk/tokens.go [hunk 3]
```diff
@@ -2,50 +2,52 @@ package clerk
...
-	FamilyName          string `json:"family_name,omitempty"`
-	PreferredUsername   string `json:"preferred_username,omitempty"`
-	Gender              string `json:"gender,omitempty"`
-	Birthdate           string `json:"birthdate,omitempty"`
-	Email               string `json:"email,omitempty"`
-	EmailVerified       bool   `json:"email_verified,omitempty"`
-	PhoneNumber         string `json:"phone_number,omitempty"`
-	PhoneNumberVerified bool   `json:"phone_number_verified,omitempty"`
+type SessionClaims struct {
+	jwt.Claims
```
Comment (agis) [inline clerk/tokens.go:23]: 🔧 Why don't we specify explicitly only the ones that the session tokens contain?

File: clerk/tokens.go [hunk 4]
```diff
@@ -2,50 +2,52 @@ package clerk
...
 
-type Claims struct {
+var (
+	standardClaimsKeys = []string{"iss", "sub", "aud", "exp", "nbf", "iat", "jti"}
+)
+
+type TokenClaims struct {
 	jwt.Claims
-	UserClaims
+	Extra map[string]interface{}
```
Comment (agis) [inline clerk/tokens.go:19]: 🔧 Instead of using an extra field, can't we just have this be a `map`? Since this is for decoding any JWT really..

Comment (chanioxaris) [reply clerk/tokens.go:19]: I decided to embed the standard claims, just to make users life easier so they don't need to check if exists in the map

File: clerk/tokens.go (hunk 3 above)
Comment (chanioxaris) [reply clerk/tokens.go:23]: Will change it accordingly to only contain the session token claims

Comment (chanioxaris) [reply clerk/tokens.go:23]: Actually by not using the `jwt.Claims` we will have to implement the validation on our own and we won't leverage the package's helper method `ValidateWithLeeway`

Comment (agis) [reply clerk/tokens.go:23]: Cool, let's leave it as is then.

File: clerk/tokens.go (hunk 4 above)
Comment (agis) [reply clerk/tokens.go:19]: Fair point!

File: clerk/tokens_test.go [hunk 5]
```diff
@@ -16,19 +16,38 @@ import (
 )
 
 var (
-	dummyClaims = Claims{
+	dummyTokenClaims = map[string]interface{}{
+		"iss":     "issuer",
+		"sub":     "subject",
+		"aud":     []string{"clerk"},
```
Comment (agis) [inline clerk/tokens_test.go:22]: Remove this

**Review by agis — APPROVED** [review]

**Pull request merged**